rsciio/mrc/_api.py

```diff
@@ -166,59 +168,110 @@ def read_de_metadata_file(filename, navigation_shape=None):
             value = value.strip()
             original_metadata[key] = value
 
-    r = int(original_metadata["Scan - Repeats"])
-    x = int(original_metadata["Scan - Size X"])
-    y = int(original_metadata["Scan - Size Y"])
-    pr = int(original_metadata["Scan - Point Repeat"])
-
-    if navigation_shape is not None:
-        shape = navigation_shape[::-1]
-        axes = []
-        for i, s in enumerate(shape):
-            axes_dict = {
-                "name": "",
-                "size": s,
-                "units": "",
-                "navigate": True,
-                "index_in_array": i,
-            }
-            axes.append(axes_dict)
-
+    # -1 -> Not read from TEM Channel 0 -> TEM 1 -> STEM
+    in_stem_mode = int(original_metadata.get("Instrument Project TEMorSTEM Mode", -1))
+    scanning = original_metadata.get("Scan - Enable", "Disable") == "Enable"
+    raster = original_metadata.get("Scan - Type", "Raster") == "Raster"
+    if not raster:  # pragma: no cover
+        _logger.warning(
+            "Non-raster scans are not fully supported yet. Please raise an issue on GitHub"
+            " if you need this feature."
+        )
+    if in_stem_mode == -1:
```

Comment:
Regarding my previous comment on adding information in file specification, I was thinking of this type of metadata, which are not documented in the MRC 2014 format.
Then these can be used for the review! :)

This doesn't need to be in this PR but longer term, we should aim to document format specification for the community. For example, in the past, the TIA-SER format was very documented online (the link is most likely still in the code) but this is not available anymore and this information is possibly lost or not openly accessible.

Reply:
That is a good point, although the metadata there is basically whatever information we have from the microscope/ our camera.  I'm working to standardize it a bit and add some documentation, but I might do that a bit later.

We can also write `hspy` which in the longer term is a bit more desirable with the eventual goal to support writing into `.zspy`.